tests/integration/preprocessor/_io/test_concatenate.py

```diff
@@ -14,34 +16,174 @@ class TestConcatenate(unittest.TestCase):
+        """Test a more generic case."""
+        self._add_cube([65., 75.], [3., 200.])
+        self._add_cube([65., 75.], [1000., 7000.])
+        concatenated = _io.concatenate(self.raw_cubes)
+        self.assertTrue(np.allclose(
+            concatenated.coord('time').points,
+            np.array([1., 2., 3., 4., 5., 6., 1000., 7000.])
+        ))
+
+    def test_concatenate_with_overlap_same_start(self):
+        """Test a more generic case."""
+        cube1 = self.raw_cubes[0]
+        raw_cubes = [cube1, ]
+        time_coord = DimCoord(
+            [1., 7.], var_name='time', standard_name='time',
+            units='days since 1950-01-01'
+        )
+        raw_cubes.append(
+            Cube([33., 55.],
+                 var_name='sample',
+                 dim_coords_and_dims=((time_coord, 0), )
+                 )
+        )
+        concatenated = _io.concatenate(raw_cubes)
+        self.assertTrue(np.allclose(
+            concatenated.coord('time').points,
+            np.array([1., 7.])
+        ))
+        raw_cubes.reverse()
+        concatenated = _io.concatenate(raw_cubes)
+        self.assertTrue(np.allclose(
+            concatenated.coord('time').points,
+            np.array([1., 7.])
+        ))
+
+    def test_concatenate_with_iris_exception(self):
+        """Test a more generic case."""
+        time_coord_1 = DimCoord(
+            [1.5, 5., 7.], var_name='time', standard_name='time',
+            units='days since 1950-01-01')
+        cube1 = Cube([33., 55., 77.],
+                     var_name='sample',
+                     dim_coords_and_dims=((time_coord_1, 0), ))
+        time_coord_2 = DimCoord(
+            [1., 5., 7.], var_name='time', standard_name='time',
+            units='days since 1950-01-01')
+        cube2 = Cube([33., 55., 77.],
+                     var_name='sample',
+                     dim_coords_and_dims=((time_coord_2, 0), ))
+        cubes_single_ovlp = [cube2, cube1]
+        with self.assertRaises(Exc.ConcatenateError):
+            _io.concatenate(cubes_single_ovlp)
+
+    def test_concatenate_with_order(self):
+        """Test a more generic case."""
+        time_coord_1 = DimCoord(
+            [1.5, 2., 5., 7.], var_name='time', standard_name='time',
+            units='days since 1950-01-01')
+        cube1 = Cube([33., 44., 55., 77.],
+                     var_name='sample',
+                     dim_coords_and_dims=((time_coord_1, 0), ))
+        time_coord_2 = DimCoord(
+            [1., 2., 5., 7., 100.], var_name='time', standard_name='time',
+            units='days since 1950-01-01')
+        cube2 = Cube([33., 44., 55., 77., 1000.],
+                     var_name='sample',
+                     dim_coords_and_dims=((time_coord_2, 0), ))
+        cubes_ordered = [cube2, cube1]
+        concatenated = _io.concatenate(cubes_ordered)
+        self.assertTrue(np.allclose(
+            concatenated.coord('time').points,
+            np.array([1., 2., 5., 7., 100.])
```

Comment:
It looks like the time point at 1.5 as defined on line 112 is simply discarded, I'm not sure this is the kind of behaviour that users would find intuitive

Reply:
Yeh well, it's part of the plan :grin: